src/meta/processors/job/JobManager.cpp

```diff
@@ -144,7 +144,7 @@ void JobManager::scheduleThread() {
 
 // @return: true if all task dispatched, else false
 bool JobManager::runJobInternal(const JobDescription& jobDesc, JbOp op) {
-  std::lock_guard<std::mutex> lk(muJobFinished_);
+  std::lock_guard<std::recursive_mutex> lk(muJobFinished_);
```

Comment:
Why we need to use recursive_mutex here?

Reply:
if all balance tasks success and the job is stopped before job status updated, the balance tasks will get muJobFinished_ through call back in the same thread

Reply:
have added comment at its declare place